### ISSUE TITLE
delete terribly slow : InfluxDB v2.x

### ISSUE THREAD

**Issue by BruceJo** [influxdata/influxdb#23974]
__Steps to reproduce:__
![image](https://user-images.githubusercontent.com/37362789/206640529-557ea515-9215-4339-bfab-5e7a277c7c0e.png)

1. Ready for some large dataset(table reference↑)
2. Try to execute ```delete from myBucket.autogen.measurement_A where time <= '2022-11-02 23:13:00'``` at ```influx v1 shell(InfluxQL Shell 2.5.0)```
3. Restore data
4. Try to execute ```./influx delete --bucket myBucket/autogen --start 1970-01-01T00:00:00Z --stop 2022-11-02T23:13:00Z --predicate '_measurement="measurement_A"'``` at ```influx CLI 2.5.0```

__Environment info:__
16C32G
Docker / Ubuntu 22.04 / binary execute influxdb OSS v2.5.1

__similar existing bug report:__
https://github.com/influxdata/influxdb/issues/9636 This issue persists in version 2.x as well.

Is there a faster way to delete it?

**Comment by stdcout42** [comment]
On 2.5.1, I'm also experiencing slow delete operations.

Furthermore, tags are left in the bucket following a full bucket data delete call, and the bucket (and the DB entirely on a single occasion) becomes completely unresponsive following a delete time out. 

On a side-note: is the table you posted auto-generated?

**Comment by trylaarsdam** [comment]
Same here. I'm trying to delete a bucket used for development with quite a lot of data in it and having issues. Neither the web UI nor the CLI can delete it. They both start the request and then just hang. During this time the bucket does not respond to queries, but if influx is restarted the data still exists and is able to be queried.

**Comment by thexperiments** [comment]
any updates here? Delete for me never finishes (runs into a 20m timeout on the reverse proxy)
I'm on 2.61 and using this query from the documentation to try and empty a bucket with wrong data:
```
influx delete --bucket example-bucket \
  --start '1970-01-01T00:00:00Z' \
  --stop $(date +"%Y-%m-%dT%H:%M:%SZ")
```
I also tried to run the query locally in the influx container but canceled it after ca 45 minutes as this might be something that will happen regularly and I can't always take that much time to delete

**Comment by nt294** [comment]
Also checking in for an update, I'm getting this on 2.7.1

**Comment by capsian** [comment]
I'm running influx inside docker and i had same issue. I found this: https://www.reddit.com/r/influxdb/comments/vluf1g/cant_delete_bucket_v23

Restarting my container solved this !

**Comment by w3blogfr** [comment]
Hello,
do we have some insight about this issue? I'm facing the same issue with this version 2.6.1

```
deleteApi.delete(event.getDate().toOffsetDateTime(),
		event.getDate().toOffsetDateTime(),
		"_measurement=\"machine-state\" AND machine = \"myMachine\"",
		bucket, organization);
```
I want to delete one specific record, and it take an infinite time, I try to increase timout without any success.

Regards.

**Comment by mschroeder-fzj** [comment]
With version `2.7.12` the issue persists.

Here is what I observed:
  - Deleting a 29GB Bucket didn't finish for over 3 weeks (with `influx delete`, aborted the delete via querry after a couple of days)
  - For checks with `influxd inspect` and the data path; the write ahead log (wal), TSI and TSM didn't show any unusual behaviours or issues
  - No erros or warnings in the logs
  - Nothing unusual except that deleting a Bucket basically didn't work, same most types of querries
  - The InfluxDB ran for around 2 Month total, deleting the Bucket started after around 1 Month of uptime

Fix: Restart the InfluxDB

A single restart and the Bucket was gone in less than a minute, same for another Bucket of over 700GB.  
The fact alone that this is the "fix" should one make worry about how broken the entire InfluxDB is.

Basically in the current state of v2 it is unusable as a long running database.  
I would say there are two options:
  1. Add a CronJob or SystemD Unit that restarts the InfluxDB like once a week
  2. Use another timeseries Database like TimescaleDB or InfluxDB v3, though I wouldn't be too surprised if there are similar issues as in v2, because there are a whole bunch of fundamental issues in v2 that no database should have (like this one)

**Comment by philjb** [comment]
@mschroeder-fzj, you might be experiencing the deadlock reported here
* https://github.com/influxdata/influxdb/issues/26889

If you see it again, please try to capture the goroutine dump `host:port/debug/pprof/goroutine?debug=2"` which could help confirm it. 

The deadlock will only be resolvable via a restart. If the delete request eventually finishes, then you don't have the deadlock; depending on the delete request, it can be quite expensive to process depending on number of series impacted.
